.github/workflows/publish_image.yaml

```diff
@@ -0,0 +1,42 @@
+name: Create and publish a Docker image to ghrc.io
+
+on:
+  push
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+  IMAGE_TAG: ${{ github.sha }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tages, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}/${{ env.IMAGE_TAG }}
```

Comment:
@wenzowski This was a little different than what you suggested, the url ends up being `ghcr.io/bcgov/connectivity-intake/a27eb4c4b9c4a2996730cf8a7b8815eacd8f3e6c:61-docker-ci` Docker didn't like having 2 colons in the address

Reply:
I'd do the branch as a tag too. I wouldn't make the sha1 an image remote. We want a reference that's unique to this build just in case we have two build running in parallel attached to the branch. If the metadata action is adding the branch name as a tag then perhaps we need to pass in the tag as a separate parameter?

Reply:
Yep, my comment described the wrong execution. Tags are [added as a list](https://github.com/docker/metadata-action#tags-input).
```yaml
tags: |
  type=schedule
  type=semver,pattern={{version}}
  type=semver,pattern={{major}}.{{minor}}
  type=semver,pattern={{major}}
  type=ref,event=branch
  type=ref,event=pr
  type=sha,format=long
```

Reply:
...and it was [already doing the thing I was asking for](https://github.com/bcgov/connectivity-intake/runs/4708583619?check_suite_focus=true#step:5:706) by default
```sh
#23 pushing ghcr.io/bcgov/connectivity-intake:sha-d14d4c70847f1f8d1bbf46c2136282bf283f5e9f with docker
```
carry on...